pallets/runtime/src/bridge.rs

```diff
@@ -0,0 +1,303 @@
+        /// to test that it cannot be called from a wrong `origin`.
+        pub fn handle_change_validators(origin, new_validators: T::AccountId) -> DispatchResult {
+            let sender = ensure_signed(origin.clone())?;
+            ensure!(sender == Self::validators(), "should be called by the validator set account");
+            // Update the validator set.
+            <Validators<T>>::put(new_validators.clone());
+            // Remove the record of the ongoing proposal.
+            <ChangeValidatorsProposals<T>>::remove(&new_validators);
+            Self::deposit_event(RawEvent::ValidatorsChanged(new_validators));
+            Ok(())
+        }
+
+        /// Handles an approved bridge transaction proposal.
+        ///
+        /// NOTE: Extrinsics without `pub` are exported too. This function is declared as `pub` only
+        /// to test that it cannot be called from a wrong `origin`.
+        pub fn handle_bridge_tx(origin, bridge_tx: BridgeTx<T::AccountId>) -> DispatchResult {
+            let sender = ensure_signed(origin.clone())?;
+            ensure!(sender == Self::validators(), "should be called by the validator set account");
+            if let Some(PendingTx {
+                did,
+                bridge_tx,
+            }) = Self::issue(bridge_tx.clone())? {
+                <PendingTxs<T>>::mutate(did, |txs| txs.push(bridge_tx.clone()));
+                Self::deposit_event(RawEvent::Pending(PendingTx{
+                    did,
+                    bridge_tx: bridge_tx.clone()
+                }));
+            } else {
+                Self::deposit_event(RawEvent::Bridged(bridge_tx.clone()));
+            }
+            // Remove the record of the ongoing proposal.
+            <BridgeTxProposals<T>>::remove(&bridge_tx);
+            Ok(())
+        }
+    }
+}
+
+impl<T: Trait> Module<T> {
+    /// Issues the transacted amount to the recipient or returns a pending transaction.
+    fn issue(bridge_tx: BridgeTx<T::AccountId>) -> IssueResult<T::AccountId> {
+        let BridgeTx {
+            nonce: _,
+            recipient,
+            value,
+            tx_hash: _,
+        } = &bridge_tx;
+        let (did, account_id) = match recipient {
+            IssueRecipient::Account(account_id) => {
+                let to_key = AccountKey::try_from(account_id.clone().encode())?;
+                (
+                    <identity::Module<T>>::get_identity(&to_key),
+                    Some(account_id),
+                )
+            }
+            IssueRecipient::Identity(did) => (Some(*did), None),
+        };
+        if let Some(did) = did {
+            // Issue to an identity or to an account associated with one.
+            if <identity::Module<T>>::has_valid_kyc(did) {
+                let amount = <T as Trait>::Balance::from(*value).into();
+                let neg_imbalance = <balances::Module<T>>::issue(amount);
```

Comment:
@adamdossa this issue will deduct funds from block rewards reserver rather than increasing total supply. Is that the desired behavior?

Reply:
Interesting point - for this type of issuance we should be increasing the total supply in Polymesh rather than depleting the block reward reserve (even if the two approaches more or less amount to the same thing).